app/src/main/java/com/example/pokemon/ui/MainActivity.kt

```diff
@@ -78,14 +85,35 @@ class MainActivity : AppCompatActivity(), MainContract.MainView {
 
     }
 
+    override fun resetPokemonList(pokeList: List<Pokemon>) {
+        pokemonAdapter?.updateData(pokeList)
+    }
+
     override fun showErrorToast() {
         Toast.makeText(this, "Error calling API", Toast.LENGTH_LONG).show()
     }
 
     override fun onDestroy() {
         super.onDestroy()
+        pokemonAdapter?.setListenerToNull()
         mainPresenterImpl?.onDestroy()
     }
 
+    override fun cardOnClick(pokemon: Pokemon, position: Int) {
+        val intent = Intent(this@MainActivity, PokemonDetailActivity::class.java)
+        mainPresenterImpl?.getPokemon(position)?.let {
```

Comment:
why are we getting pokemon from presenter when we already have pokemon object coming in method ?